dimod/core/sampler.py

```diff
@@ -164,14 +165,22 @@ def samplemixinmethod(method):
     return method
 
 
-class Sampler(metaclass=SamplerABCMeta):
+class Sampler(Scoped, metaclass=SamplerABCMeta):
     """Abstract base class for dimod samplers.
 
     Provides all methods :meth:`~.Sampler.sample`, :meth:`~.Sampler.sample_ising`,
     :meth:`~.Sampler.sample_qubo` assuming at least one is implemented.
 
     Also includes utility method :meth:`~.Sampler.remove_unknown_kwargs`,
     which may be used in sample methods to handle unknown kwargs.
+
+    Default implementation of :meth:`~.Sampler.close` does nothing, so it should
+    be overridden to release scope-based sampler resources, it any.
```

Comment:
```suggestion
    be overridden to release any scope-based sampler resources.
```

typo ("it" --> "if")